packages/fxa-auth-server/scripts/cleanup-partial-firestore-customers.ts

```diff
@@ -0,0 +1,231 @@
+    );
+    // Filter uids where account does not exist
+    // Uids where account doesn't exist, should be deleted
+    const uidsForDelete = uids.filter(
+      (uid) => !accountUids.find((accountId) => accountId === uid)
+    );
+
+    return uidsForDelete;
+  }
+
+  private async callPartialDelete(uids: string[]) {
+    this.log.debug('CleanupFirestoreHelper.callPartialDelete.start', {
+      uidsForDelete: uids.length,
+    });
+    if (this.dryRun) {
+      this.log.debug('CleanupFirestoreHelper.callPartialDelete.skipped', {
+        dryRun: this.dryRun,
+        uidNumbers: uids.length,
+      });
+
+      return;
+    }
+
+    await this.cloudTaskQueue.addAll(
+      uids.map(
+        (uid) => () =>
+          this.accountDeleteManager.enqueue({
+            uid,
+            reason: AccountDeleteReasonsMap.cleanup,
+          })
+      )
+    );
+  }
+
+  async cleanup() {
+    this.log.debug('DeletePartial.start', {
+      batchSize: this.batchSize,
+      dryRun: this.dryRun,
+    });
+    let startAfter: QueryDocumentSnapshot | undefined = undefined;
+    let batchCounter = 0;
+    const hasMore = true;
+    while (hasMore) {
+      this.log.info('CleanupFirestoreHelper.cleanup', {
+        msg: `Batch ${batchCounter * this.batchSize} - ${
+          (batchCounter + 1) * this.batchSize - 1
+        }`,
+      });
+      batchCounter += 1;
+      // 1. Fetch customers from Firestore
+      const { uids: uidsFirestore, startAfter: newStartAfter } =
+        await this.fetchFirestoreCustomersUid(startAfter);
+
+      if (!newStartAfter) {
+        break;
+      } else {
+        startAfter = newStartAfter;
+      }
+
+      // 2. Find uids that need to be deleted
+      const uidsForDelete = await this.getUidsForDelete(uidsFirestore);
+
+      // 3. Enqueue uids for Cloud Task account delete
+      await this.callPartialDelete(uidsForDelete);
+    }
+  }
+}
+
+export async function init() {
+  program
+    .version(pckg.version)
+    .option(
+      '-b, --batch-size [number]',
+      'Number of customers to query from firestore at a time.  Defaults to 100.',
+      100
+    )
+    .option(
+      '-r, --rate-limit [number]',
+      'Number of tasks to add to Cloud Tasks at a time. Defaults to 10.',
+      10
+    )
+    .option(
+      '--dry-run [true|false]',
+      'Print what the script would do instead of performing the action. Defaults to true.',
+      true
+    )
+    .parse(process.argv);
+
+  const options = program.opts();
+  const batchSize = parseInt(options.batchSize);
+  const rateLimit = parseInt(options.rateLimit);
+  const isDryRun = parseDryRun(options.dryRun);
+
+  const { database: fxaDb } = await setupProcessingTaskObjects(
+    'cleanup-delete-partial-firestore'
+  );
+
+  const config = Container.get(AppConfig);
+  const statsd = Container.get(StatsD);
+  const log = Container.get(AuthLogger);
+  const pushbox = pushboxApi(log, config, statsd);
+
+  const accountDeleteManager = new AccountDeleteManager({
```

Comment:
This needs a `push` object, but you can pass an empty object for it (see other delete script).